flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/JobAutoScaler.java

```diff
@@ -30,6 +28,6 @@ public interface JobAutoScaler {
     /** Called when the custom resource is deleted. */
     void cleanup(FlinkResourceContext<?> ctx);
 
-    /** Get the current parallelism overrides for the job. */
-    Map<String, String> getParallelismOverrides(FlinkResourceContext<?> ctx);
+    /** Apply current parallelism overrides. */
+    void applyParallelismOverrides(FlinkResourceContext<?> ctx);
```

Comment:
Is this interface method really necessary? Overrides should be applied in the scale method, no?

Reply:
This is based on earlier discussion in https://issues.apache.org/jira/browse/FLINK-32589 
Overrides need to be applied separately (and it was implemented already this way I just relocated the logic) so that we do not lose them on spec changes.

Reply:
I couldn't find any discussion in that thread which would explain why the overrides couldn't be applied without the method here. Perhaps the code would need refactoring (which is done here already) to apply the scale method before reconciling other parts of the spec.